Automattic/wp-calypso
# Site Importer: Auto start when coming up from Signup

This PR is part of series to extract more focused changes out of #32449 

This diff adds the following:
* A temporary "loading" screen similar to the in-progress screens for imports while the import section properly loads.
* The ability to auto-start the import with a query parameter (`autoStartSiteImport`) if there's a need.

At the start the imports won't auto start, unless specified with the query parameter mentioned above in order to preserve the current behavior and we work out proper design/flow.

# To test Signup flow - autostart
* Checkout branch (won't work on Calypso.live as it requires code edit)
* Go to `client/signup/config/flows-pure.js` -> function `importDestination`
* Change `autoStartSiteImport` to `1` 
* Build Calypso
* Go to Create New site on the local Calypso instance
* Choose `Already have a site?` option
* Follow the flow and try with a Wix and a GoDaddy site
* Finish signup
* Make sure the import auto starts and completes

# To test Signup flow - no autostart
* Checkout branch or use Calypso.live link
* Go to Create New site
* Choose `Already have a site?` option
* Follow the flow and try with a Wix and a GoDaddy site
* Finish signup
* Make sure the import DOES NOT auto start

# To test normal Import flow

* Checkout branch or use Calypso.live link
* Go to Imports
* Choose Wix
* Enter a site URL
* Continue through the Preview step
* Make sure the import auto progresses after clicking Next on the Preview screen
* Make sure there are no JS errors in the Developer console.
* Repeat for a GoDaddy site
